File: python/mp3/encode_wav_to_mp3.py
################################################################################
# wav ファイルを mp3 にエンコードします。
################################################################################

import glob
import json
import os
import subprocess

# 作業ディレクトリのパス
WORKING_DIRECTORY_PATH = os.path.dirname(os.path.abspath(__file__))

# 設定をロード
settings = json.load(open(f"{WORKING_DIRECTORY_PATH}\\settings.json", "r"))

# 設定された引数を配列に変換
lame_args = settings["lame_arg"].split()

# 作業ディレクトリと同階層にあるすべての wav ファイルに対して処理を行う
for wav in glob.glob(f"{WORKING_DIRECTORY_PATH}\\*.wav"):
    dest = f"{os.path.splitext(wav)[0]}.mp3"

    command = []
    command.append(settings["lame_path"])
    command.extend(lame_args)
    command.append(wav)
    command.append(dest)

    subprocess.call(command)

File: ruby/mp3/encode_wav_to_mp3.rb
# encoding: utf-8
################################################################################
# wav ファイルを mp3 にエンコードします。
################################################################################

require "json"
require "open3"

# 作業ディレクトリのパス
WORKING_DIRECTORY_PATH = __dir__.encode("UTF-8").freeze

# 設定をロード
begin
  config = File.open("#{WORKING_DIRECTORY_PATH}/config.json") { |f| JSON.load(f) }
rescue
  puts "config.json を読み込めませんでした。"
  exit
end

# 設定された引数を配列に変換
lame_args = config["lame_arg"].split(" ")

# エンコードに失敗したファイルを記憶しておく配列
failed_files = []

# 作業ディレクトリと同階層にあるすべての wav ファイルに対して処理を行う
Dir.glob("#{WORKING_DIRECTORY_PATH}/*.wav").each do |wav|
  dest = "#{File.dirname(wav)}/#{File.basename(wav, ".wav")}.mp3"

  command = []
  command.push(config["lame_path"])
  command.concat(lame_args)
  command.push(wav)
  command.push(dest)

  print "Encoding #{wav} ... "

  # エンコードを実行
  o, e, s = Open3.capture3(*command)

  if s.exited?
    puts "OK"
  else
    puts "NG"
    failed_files.push(wav)
  end
end

if failed_files.empty?
  puts "すべてのファイルが正常にエンコードされました。"
else
  puts "以下のファイルのエンコードに失敗しました："
  puts failed_files
end
